x11: build fails if Xfixes isn't available since commit 4fd77811 (PR #8472)

See https://github.com/libsdl-org/sdl2-compat/actions/runs/7136397469/job/19440224526 :

```
  /tmp/setupsdl/62679c63e4a5f564085d43fa117cd506ee48968f54745f84b861433abc946451/source/src/video/x11/SDL_x11events.c: In function ‘X11_DispatchEvent’:
  /tmp/setupsdl/62679c63e4a5f564085d43fa117cd506ee48968f54745f84b861433abc946451/source/src/video/x11/SDL_x11events.c:1351:22: error: ‘SDL_WindowData’ has no member named ‘skip_size_count’
   1351 |             if (!data->skip_size_count) {
        |                      ^~
  /tmp/setupsdl/62679c63e4a5f564085d43fa117cd506ee48968f54745f84b861433abc946451/source/src/video/x11/SDL_x11events.c:1356:21: error: ‘SDL_WindowData’ has no member named ‘skip_size_count’
   1356 |                 data->skip_size_count--;
        |                     ^~
  /tmp/setupsdl/62679c63e4a5f564085d43fa117cd506ee48968f54745f84b861433abc946451/source/src/video/x11/SDL_x11events.c:1631:25: error: ‘SDL_WindowData’ has no member named ‘pending_operation’
   1631 |                     data->pending_operation &= ~X11_PENDING_OP_FULLSCREEN;
        |                         ^~
  /tmp/setupsdl/62679c63e4a5f564085d43fa117cd506ee48968f54745f84b861433abc946451/source/src/video/x11/SDL_x11events.c:1631:49: error: ‘X11_PENDING_OP_FULLSCREEN’ undeclared (first use in this function)
   1631 |                     data->pending_operation &= ~X11_PENDING_OP_FULLSCREEN;
        |                                                 ^~~~~~~~~~~~~~~~~~~~~~~~~
  /tmp/setupsdl/62679c63e4a5f564085d43fa117cd506ee48968f54745f84b861433abc946451/source/src/video/x11/SDL_x11events.c:1631:49: note: each undeclared identifier is reported only once for each function it appears in
  /tmp/setupsdl/62679c63e4a5f564085d43fa117cd506ee48968f54745f84b861433abc946451/source/src/video/x11/SDL_x11events.c:1635:57: error: ‘SDL_WindowData’ has no member named ‘requested_fullscreen_mode’
   1635 |                             const SDL_bool commit = data->requested_fullscreen_mode.displayID == 0 ||
        |                                                         ^~
  /tmp/setupsdl/62679c63e4a5f564085d43fa117cd506ee48968f54745f84b861433abc946451/source/src/video/x11/SDL_x11events.c:1636:109: error: ‘SDL_WindowData’ has no member named ‘requested_fullscreen_mode’
   1636 |                                                     SDL_memcmp(&data->window->current_fullscreen_mode, &data->requested_fullscreen_mode, sizeof(SDL_DisplayMode)) != 0;
        |                                                                                                             ^~
  /tmp/setupsdl/62679c63e4a5f564085d43fa117cd506ee48968f54745f84b861433abc946451/source/src/video/x11/SDL_x11events.c:1656:29: error: ‘SDL_WindowData’ has no member named ‘skip_size_count’
   1656 |                         data->skip_size_count = 1;
        |                             ^~
  /tmp/setupsdl/62679c63e4a5f564085d43fa117cd506ee48968f54745f84b861433abc946451/source/src/video/x11/SDL_x11events.c:1660:25: error: ‘SDL_WindowData’ has no member named ‘pending_operation’
   1660 |                     data->pending_operation &= ~X11_PENDING_OP_MAXIMIZE;
        |                         ^~
  /tmp/setupsdl/62679c63e4a5f564085d43fa117cd506ee48968f54745f84b861433abc946451/source/src/video/x11/SDL_x11events.c:1660:49: error: ‘X11_PENDING_OP_MAXIMIZE’ undeclared (first use in this function)
   1660 |                     data->pending_operation &= ~X11_PENDING_OP_MAXIMIZE;
        |                                                 ^~~~~~~~~~~~~~~~~~~~~~~
  /tmp/setupsdl/62679c63e4a5f564085d43fa117cd506ee48968f54745f84b861433abc946451/source/src/video/x11/SDL_x11events.c:1662:29: error: ‘SDL_WindowData’ has no member named ‘pending_operation’
   1662 |                         data->pending_operation &= ~X11_PENDING_OP_RESTORE;
        |                             ^~
  /tmp/setupsdl/62679c63e4a5f564085d43fa117cd506ee48968f54745f84b861433abc946451/source/src/video/x11/SDL_x11events.c:1662:53: error: ‘X11_PENDING_OP_RESTORE’ undeclared (first use in this function)
   1662 |                         data->pending_operation &= ~X11_PENDING_OP_RESTORE;
        |                                                     ^~~~~~~~~~~~~~~~~~~~~~
  /tmp/setupsdl/62679c63e4a5f564085d43fa117cd506ee48968f54745f84b861433abc946451/source/src/video/x11/SDL_x11events.c:1669:25: error: ‘SDL_WindowData’ has no member named ‘pending_operation’
   1669 |                     data->pending_operation &= ~X11_PENDING_OP_MINIMIZE;
        |                         ^~
  /tmp/setupsdl/62679c63e4a5f564085d43fa117cd506ee48968f54745f84b861433abc946451/source/src/video/x11/SDL_x11events.c:1669:49: error: ‘X11_PENDING_OP_MINIMIZE’ undeclared (first use in this function)
   1669 |                     data->pending_operation &= ~X11_PENDING_OP_MINIMIZE;
        |                                                 ^~~~~~~~~~~~~~~~~~~~~~~
  /tmp/setupsdl/62679c63e4a5f564085d43fa117cd506ee48968f54745f84b861433abc946451/source/src/video/x11/SDL_x11events.c:1673:25: error: ‘SDL_WindowData’ has no member named ‘pending_operation’
   1673 |                     data->pending_operation &= ~X11_PENDING_OP_RESTORE;
        |                         ^~
```

CC: @Kontrabant

--- a/src/video/x11/SDL_x11window.h
+++ b/src/video/x11/SDL_x11window.h
@@ -78,7 +78,7 @@ struct SDL_WindowData
     SDL_bool pointer_barrier_active;
     PointerBarrier barrier[4];
     SDL_Rect barrier_rect;
-
+#endif /* SDL_VIDEO_DRIVER_X11_XFIXES */
     SDL_Rect expected;
     SDL_DisplayMode requested_fullscreen_mode;
 
@@ -94,7 +94,6 @@ struct SDL_WindowData
     SDL_bool initial_border_adjustment;
     SDL_bool window_was_maximized;
     int skip_size_count;
-#endif /* SDL_VIDEO_DRIVER_X11_XFIXES */
     SDL_HitTestResult hit_test_result;
 };
 

